packages/dapp/src/components/OnRamper/OnRamper.tsx

```diff
@@ -34,18 +34,25 @@ export const OnRamper = () => {
     borderRadius: 0.75,
     wgBorderRadius: 1.5,
   };
-  const onRamperSrc = `https://buy.onramper.com/?themeName=${
-    onRamperConfig.themeName
-  }&defaultCrypto=${onRamperConfig.defaultCrypto}
-  &containerColor=${onRamperConfig.containerColor}&primaryColor=${
-    onRamperConfig.primaryColor
-  }&secondaryColor=${onRamperConfig.secondaryColor}&cardColor=${
-    onRamperConfig.cardColor
-  }&primaryTextColor=${onRamperConfig.primaryTextColor}&secondaryTextColor=${
-    onRamperConfig.secondaryTextColor
-  }&borderRadius=${onRamperConfig.borderRadius}&wgBorderRadius=${
-    onRamperConfig.wgBorderRadius
-  }?apiKey=${import.meta.env.VITE_ONRAMPER_API_KEY}`;
+
+  const searchParams = new URLSearchParams();
+  searchParams.append('themeName', onRamperConfig.themeName);
+  searchParams.append('defaultCrypto', onRamperConfig.defaultCrypto);
+  searchParams.append('containerColor', onRamperConfig.containerColor);
+  searchParams.append('primaryColor', onRamperConfig.primaryColor);
+  searchParams.append('secondaryColor', onRamperConfig.secondaryColor);
+  searchParams.append('cardColor', onRamperConfig.cardColor);
+  searchParams.append('primaryTextColor', onRamperConfig.primaryTextColor);
+  searchParams.append('secondaryTextColor', onRamperConfig.secondaryTextColor);
+  searchParams.append('borderRadius', onRamperConfig.borderRadius.toString());
+  searchParams.append(
+    'wgBorderRadius',
+    onRamperConfig.wgBorderRadius.toString(),
+  );
+  searchParams.append('apiKey', import.meta.env.VITE_ONRAMPER_API_KEY);
```

Comment:
It should be possible to simplify it to

`const searchParams = new URLSearchParams(onRamperConfig);`

Please check types, maybe conversion of some props to strings will be required 🙂

Reply:
@chybisov: I have re-iterated on it. What do you think about the latest changes - basically looping through the existing onRamperConfig.